plugins/basic/python/tk_maya_basic/plugin_logic.py

```diff
@@ -43,6 +44,9 @@
 # Use a custom logging handler to display messages in Maya script editor.
 standalone_logger.addHandler(plugin_logging.PluginLoggingHandler(manifest.name))
 
+# Main progress bar used when bootstrapping the toolkit and engine.
+main_progress_bar = None
```

Comment:
do we need to keep a global around for this? Maya already does this for us? In my experience, I tend to try as few 'pointers' into maya as possible. Would it make sense to do 

```
def get_maya_main_progress_bar():
    return pm.ui.MainProgressBar(minValue=0, maxValue=100, interruptable=False)
```

What happens if the plugin is downloading stuff from the app store and the user keeps working, doing other operations which also utilises the main progress bar? For this reason, I think it may be safer if the bootstrap is getting the progress bar every time.